### DISCUSSION TITLE
Live reload with common email layout component

### DISCUSSION THREAD

**Discussion by joranb** [shellscape/jsx-email#268] [Q&A] [answered]
Hi, I am using jsx-email with keycloakify-emails to generate Keycloak email templates. I only have 1 problem. When following the guide there is an emails folder with a layout.tsx, and a templates folder. I have generated password-reset.tsx inside that folder, and live reload when using preview works great for that file. But not when I make changes to layout.tsx in the parent directory. So I basically have to make my common email layout blind. Is there a way around this? Would not have been a problem if my email layout was already finished, since then I would only make changes to the template files anyway. Or maybe I am just a shitty coder, and it is not a problem for you guys since you just compile it in your heads. 

Tried putting the layout file inside the templates folder, but that fails since it seems to only work for .tsx-files which export Template. Have tried searching the docs for a solution but have not succeeded yet. Any help would be appreciated! 

Edit:
Found a hack, exported a template inside my layout file using the layout, then I did not get an error when running preview on the emails folder. And live reload works for the layout, for that dummy template, but not for the template in templates folder. But that is ok, since I now got live reload when designing the layout for the dummy template.

**Comment by shellscape** [comment, marked as answer]
This changed in 2.4.2 with fixes to the watcher to prevent recursive watching problems with Next.js apps. Always important to read the `CHANGELOG` for any npm package you're using. https://github.com/shellscape/jsx-email/blob/main/packages/jsx-email/CHANGELOG.md#bugfixes-4

There are more details in the associated PR for that change.

We're working on making import/dependency watching and reload working again. I don't have an ETA for it, but it will be before the end of the year. 

In the mean time, you'll have to restart the preview when import/dependency files change. We don't normally rollback things like this, but the implementation that was in place was causing more problems that it was worth.

**Reply by joranb** [reply]
Thank you for the swift reply, that explains it. Did not think of checking the changelog. Great job with this by the way, makes it a lot easier!
